src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/Processors/LedgerTransactionMarkers/ResourceBalanceChangeProcessor.cs

```diff
@@ -0,0 +1,115 @@
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using RadixDlt.CoreApiSdk.Model;
+using RadixDlt.NetworkGateway.Abstractions;
+using RadixDlt.NetworkGateway.Abstractions.Network;
+using RadixDlt.NetworkGateway.PostgresIntegration.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace RadixDlt.NetworkGateway.PostgresIntegration.LedgerExtension.Processors.LedgerTransactionMarkers;
+
+internal class ResourceBalanceChangeProcessor : ITransactionMarkerProcessor, ITransactionProcessor
+{
+    private readonly ProcessorContext _context;
+    private readonly ReferencedEntityDictionary _referencedEntities;
+    private readonly Dictionary<long, HashSet<long>> _resourcesWithBalanceChanges = new();
+
+    public ResourceBalanceChangeProcessor(ProcessorContext context, ReferencedEntityDictionary referencedEntities)
+    {
+        _context = context;
+        _referencedEntities = referencedEntities;
+    }
+
+    public void VisitTransaction(CommittedTransaction transaction, long stateVersion)
+    {
+        var fungibleResourceBalanceChanges = transaction.BalanceChanges.FungibleEntityBalanceChanges
+            .SelectMany(x => x.NonFeeBalanceChanges)
+            .Select(x => _referencedEntities.Get((EntityAddress)x.ResourceAddress).DatabaseId)
```

Comment:
I assume we're guaranteed to have loaded these as referenced entities via handling the vault state changes or something? Maybe worth a comment if it's not immediately obvious to you?

Reply:
yes it's guaranteed, that's basic gateway concept it'd mean we have to comment all places where we use that (50+ places). I'd prefer not to add it.

Reply:
Well what I mean is - do we explicitly load them from the `FungibleEntityBalanceChanges` etc?
I assume not? So it's implicit that they're loaded in (via some other registrations of these addresses), and a refactor which removed those could theoretically break this. I guess if we have tests, we'll catch this.
When I did Olympia; each processor had a pre-process step (where e.g. it would register things to load from the DB) and then a process step, and it was expected that it would keep these in parallel. Here it's implicit. I think that's okay.
Happy for us not to document if you think it's sufficiently safe/obvious.

Reply:
yes we load them explicitly here by using `IEntityAddressPointer` interface. All addresses returned by that interface are fetched from db. https://github.com/radixdlt/babylon-gateway/blob/main/src/RadixDlt.CoreApiSdk/Model/CommittedTransactionBalanceChanges.cs

Reply:
Ahh nice! 👍